kura/org.eclipse.kura.linux.bluetooth/src/main/java/org/eclipse/kura/linux/bluetooth/le/BluetoothLeScanner.java

```diff
@@ -173,6 +191,20 @@ public void processBTSnoopRecord(byte[] record) {
 
 		try {
 			
+			// Extract raw advertisement data
+			BluetoothAdvertisementData bAdData = BluetoothUtil.parseLEAdvertisement(record);
+			
+			// Notify advertisement listeners
+			if (bAdData != null) {
```

Comment:
couldn't `bAdData != null` and `m_advertisement_listener != null` be collapsed in a single if?